File: backend/bookingmovieticketapp/src/main/java/com/project/bookingmovieticketapp/Models/ShowTime.java
package com.project.bookingmovieticketapp.Models;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "showtimes")
@Builder
public class ShowTime {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @ManyToOne
    @JoinColumn(name = "movieid")
    private Movie movie;

    @ManyToOne
    @JoinColumn(name = "roomid")
    private Room room;

    @Column(name = "showdate",nullable = false)
    private LocalDate showdate;

    @Column(name = "starttime",nullable = false)
    private LocalDateTime starttime;

    @Column(name = "price",nullable = false)
    private int price;
}

File: frontendapp/lib/models/showtime.dart
class Showtime {
  final int? id;
  final int movieId;
  final int roomId;
  final DateTime showDate;
  final DateTime startTime;
  final int price;

  Showtime({
    this.id,
    required this.movieId,
    required this.roomId,
    required this.showDate,
    required this.startTime,
    required this.price,
  });

  factory Showtime.fromJson(Map<String, dynamic> json) => Showtime(
    id: json['id'],
    movieId: json['movieId'],
    roomId: json['roomId'],
    showDate: DateTime.parse(json['showDate']),
    startTime: DateTime.parse(json['startTime']),
    price: json['price'],
  );
} 
